### PR TITLE
Use fmod for navigation mesh border size warning

### PR DESCRIPTION
**Pull request by smix8**
Uses fmod for checking that navigation mesh `border_size` matches the `cell_size` as a multiple of cell size is also valid.

Also removes `is_equal_approx()` in this case as having the numbers not match exactly will cause edge seams on very small cell sizes which goes against the idea of using a border size in the first place.

### CODE REVIEW

File: modules/navigation/3d/nav_mesh_generator_3d.cpp [hunk 1]
```diff
@@ -700,7 +700,7 @@ void NavMeshGenerator3D::generator_bake_from_source_geometry_data(Ref<Navigation
 	cfg.detailSampleDist = MAX(p_navigation_mesh->get_cell_size() * p_navigation_mesh->get_detail_sample_distance(), 0.1f);
 	cfg.detailSampleMaxError = p_navigation_mesh->get_cell_height() * p_navigation_mesh->get_detail_sample_max_error();
 
-	if (p_navigation_mesh->get_border_size() > 0.0 && !Math::is_equal_approx(p_navigation_mesh->get_cell_size(), p_navigation_mesh->get_border_size())) {
+	if (p_navigation_mesh->get_border_size() > 0.0 && Math::fmod((float)p_navigation_mesh->get_border_size(), (float)p_navigation_mesh->get_cell_size()) != 0.0) {
```
Comment (aaronfranke) [inline modules/navigation/3d/nav_mesh_generator_3d.cpp:703]: The casts are redundant because both of these already return `float`.

Part of me wonders if this should be `real_t` to support navigation meshes that are bigger than a few kilometers, but this should be left to another PR. For this PR, I think the casts would only be harmful if we ever do change the return types to `real_t`, so I would just remove the casts.

**Review by aaronfranke — APPROVED** [review]
I'm not familiar with this code but it looks good to me, and it meets the stated goals in the PR description.

**Comment by akien-mga** [comment]
Thanks!
